### PR TITLE
user/snapper: new package

### PR DESCRIPTION
**Pull request by tulilirockz**

### CODE REVIEW

File: user/snapper/patches/btrfs.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include <unistd.h>
+-#ifdef HAVE_LIBBTRFS
+ #include <btrfs/ioctl.h>
+ #include <btrfs/send-utils.h>
+-#else
+-#include <linux/btrfs.h>
+-#endif
+-#ifdef HAVE_LIBBTRFSUTIL
```
Comment (nekopsykose) [inline user/snapper/patches/btrfs.patch:13]: not sure what this fixes, this is already the path taken if these HAVE_ are set and they should be, if they're not then the build system has to be fixed instead..?

File: user/snapper/patches/ncurses.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+diff --git a/client/utils/Makefile.am b/client/utils/Makefile.am
+index e6a9968..aeeea5a 100644
+--- a/client/utils/Makefile.am
++++ b/client/utils/Makefile.am
+@@ -20,4 +20,4 @@ libutils_la_SOURCES =					\
+ 	CsvFormatter.cc	    	CsvFormatter.h		\
+ 	JsonFormatter.cc    	JsonFormatter.h
+ 
+-libutils_la_LIBADD = ../../snapper/libsnapper.la -ltinfo
++libutils_la_LIBADD = ../../snapper/libsnapper.la -lncurses
```
Comment (nekopsykose) [inline user/snapper/patches/ncurses.patch:10]: this isn't needed, libtinfo compat symlinks exist, they're just in ncurses-libtinfo-devel for some reason, i'll put them in the main one

File: user/snapper/template.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    "boost-devel",
+    "dbus-devel",
+    "e2fsprogs-devel",
+    "json-c-devel",
+    "libbtrfs-devel",
+    "libbtrfsutil-devel",
+    "libmount-devel",
+    "libxml2-devel",
+    "linux-pam-devel",
+    "ncurses-devel",
```
Comment (nekopsykose) [inline user/snapper/template.py:36]: stray newlines

File: user/snapper/template.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    self.install_file("data/sysconfig.snapper", "etc/snapper", name="snapper")
+    self.install_service(self.files_path / "snapper")
+    self.install_tmpfiles(self.files_path / "tmpfiles.conf")
+    self.uninstall("usr/lib/snapper/testsuite")
+    self.uninstall("usr/lib/snapper/systemd-helper")
+
+
+@subpackage("snapper-libs")
+def _(self):
+    self.install_if = [self.parent]
```
Comment (nekopsykose) [inline user/snapper/template.py:57]: the parent links to the libs already so this install_if doesn't do anything

but there's also no reason to split them that i see

File: user/snapper/template.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    return self.default_libs()
+
+
+@subpackage("pam_snapper")
+def _(self):
+    self.install_if = [self.parent]
```
Comment (nekopsykose) [inline user/snapper/template.py:63]: don't need this subpackage either

File: user/snapper/patches/btrfs.patch (hunk 1 above)
Comment (nekopsykose) [reply user/snapper/patches/btrfs.patch:13]: i guess this is easier than making the build system define HAVE_LIBBTRFS, but the latter is defined ok afaict

File: user/snapper/template.py (hunk 5 above)
Comment (nekopsykose) [reply user/snapper/template.py:63]: but you do need a -devel one

File: user/snapper/patches/btrfs.patch (hunk 1 above)
Comment (tulilirockz) [reply user/snapper/patches/btrfs.patch:13]: apparently these HAVE_s arent being set properly. ive tried doing a ton of stuff and it wouldnt fix this damn thing

Comment (nekopsykose) [reply user/snapper/patches/btrfs.patch:13]: yeah, the former is not checked at all :)

File: user/snapper/template.py (hunk 4 above)
Comment (tulilirockz) [reply user/snapper/template.py:57]: from what I know, transactional-update depends on libsnapper but not the snapper CLI... althought that probably would even work on chimera, right? ill just merge it into the base package

**Comment by tulilirockz** [comment]
oh my god
what

**Comment by tulilirockz** [comment]
what happened

**Comment by nekopsykose** [comment]
you rebased something wrong

**Comment by tulilirockz** [comment]
better now 👍

**Comment by tulilirockz** [comment]
patched cron script cuz without that it wont get recognized by run-parts

**Pull request merged**